wordpress-mobile/WordPress-iOS
# 16.8.1 Hot Fix

This is a PR to a hot fix branch to fix an issue with Stories blocks overwriting the contents of a post.

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/pull/3246

Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3246
Gutenberg PR: https://github.com/WordPress/gutenberg/pull/29734

To test:
* Add some blocks to a post.
* Add a Story block.
* Add a media item.
* Tap on the right arrow button to continue.
* Observe that the previous added blocks are **not** removed.

## Changes

* Uses a new `replace(blockID: blockID, content: content)` in Gutenberg to replace only the block's content instead of handling the contents of the entire post.
* Removes some unnecessary Alerts when posts fail to complete uploads. These can be retried later on the Post List screen.

## Regression Scenarios

### New Story Post

* Tap the FAB
* Select "Story post"
* Select multiple media from the Photo or WordPress libraries
* Add more media by tapping the + button in the bottom right
* Take a photo with the camera
* Publish the post by tapping the arrow in the upper right, filling out the sheet, and hitting Publish
* Ensure that the story posts and contents are visible in the post

### New Post with Story block

* Create a new Blog Post
* Add a title
* Add a paragraph with text
* Add a Story block
* Click "Add Media"
* Add media from your library or camera
* Save by tapping the arrow icon in the upper right
* Ensure that the block updates with the new media
* Tap the Publish button to publish the post

### Editing an existing post

* Edit an existing Story post
* Tap the "Edit" icon in the top right of the Story block
* Ensure that the media is loaded
* Ensure that overlays (text) is editable if the story was posted from the device
* Make sure that editing and saving the contents of the story causes the block to be updated
* Make sure that Updating the post saves those edits and updates the preview for the post

### Discarding edits to a post

* Edit an existing Story post
* Tap the "Edit" icon in the top right of the Story block
* Ensure that the media is loaded
* Ensure that overlays (text) is editable if the story was posted from the device
* Make changes to the post
* Save the changes by tapping the arrow in the upper right
* Discard the post changes by tapping the X in the top left in the Gutenberg editor
* Make sure the post isn't updated

### Discarding edits in Kanvas

* Edit an existing Story post
* Tap the "Edit" icon in the top right of the Story block
* Ensure that the media is loaded
* Ensure that overlays (text) is editable if the story was posted from the device
* Make changes to the post
* Tap the X in the top left.
* Make sure a dialog comes up to confirm you would like to discard the edits
* Hit "Discard"
* Ensure the block is not updated and no changes have been made to the post.

PR submission checklist:
